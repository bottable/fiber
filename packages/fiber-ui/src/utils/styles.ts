import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
