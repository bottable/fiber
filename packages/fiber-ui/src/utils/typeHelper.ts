export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
