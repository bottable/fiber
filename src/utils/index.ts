export * from './typeHelper'
export * from './mergeProps'
