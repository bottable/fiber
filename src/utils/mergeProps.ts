import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
