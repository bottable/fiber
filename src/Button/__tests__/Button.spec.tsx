import { Button } from '..'

import React from 'react'
import { render } from 'test/utils'
import DeleteIcon from '@material-ui/icons/Delete'

describe('Button', () => {
  it('renders correctly', () => {
    const ButtonGroup = render(
      <>
        <div>
          <Button type='primary' disabled>
            Primary Button
          </Button>
          <Button disabled>Default Button</Button>
          <Button type='dashed' disabled>
            Dashed Button
          </Button>
          <Button type='text' disabled>
            Text Button
          </Button>
          <Button type='link' disabled>
            Link Button
          </Button>
        </div>
        <div style={{ backgroundColor: 'rgb(190, 200, 200)', paddingTop: 12 }}>
          <Button type='primary' ghost>
            Primary Button
          </Button>
          <Button ghost>Default Button</Button>
          <Button type='dashed' ghost>
            Dashed Button
          </Button>
          <Button type='text' ghost>
            Text Button
          </Button>
          <Button type='link' ghost>
            Link Button
          </Button>
        </div>
        <div
          style={{
            width: 200,
            border: '1px solid black',
            padding: 25,
            margin: 10
          }}
        >
          <Button type='primary' block>
            Primary Button
          </Button>
          <Button block>Default Button</Button>
          <Button type='dashed' block>
            Dashed Button
          </Button>
          <Button type='text' block>
            Text Button
          </Button>
          <Button type='link' block>
            Link Button
          </Button>
        </div>
        <div>
          <Button icon={<DeleteIcon />} />
          <Button icon={<DeleteIcon />} shape='circle' />
          <Button icon={<DeleteIcon />} shape='round' />
        </div>
        <div>
          <Button icon={<DeleteIcon />} size='lg'>
            Trash
          </Button>
          <Button icon={<DeleteIcon />}>Trash</Button>
          <Button icon={<DeleteIcon />} size='sm'>
            Trash
          </Button>
        </div>
      </>
    )

    expect(ButtonGroup).toMatchSnapshot()
  })
})
