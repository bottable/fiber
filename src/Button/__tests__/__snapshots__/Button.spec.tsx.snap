// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
Array [
  <div>
    <button
      className="sc-AxiKw sc-AxheI iUUfME"
      disabled={true}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Primary Button
      </span>
    </button>
    <button
      className="sc-AxiKw sc-Axmtr fbANLJ"
      disabled={true}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Default Button
      </span>
    </button>
    <button
      className="sc-AxiKw sc-AxmLO dnARkq"
      disabled={true}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Dashed Button
      </span>
    </button>
    <button
      className="sc-AxiKw sc-fzozJi fBCfGu"
      disabled={true}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Text Button
      </span>
    </button>
    <button
      className="sc-AxiKw sc-fzoLsD hnYQsF"
      disabled={true}
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Link Button
      </span>
    </button>
  </div>,
  <div
    style={
      Object {
        "backgroundColor": "rgb(190, 200, 200)",
        "paddingTop": 12,
      }
    }
  >
    <button
      className="sc-AxiKw sc-AxheI czYOWC"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Primary Button
      </span>
    </button>
    <button
      className="sc-AxiKw sc-Axmtr kVmKXA"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Default Button
      </span>
    </button>
    <button
      className="sc-AxiKw sc-AxmLO bHfEFN"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Dashed Button
      </span>
    </button>
    <button
      className="sc-AxiKw sc-fzozJi dDKydC"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Text Button
      </span>
    </button>
    <button
      className="sc-AxiKw sc-fzoLsD bAkHPd"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Link Button
      </span>
    </button>
  </div>,
  <div
    style={
      Object {
        "border": "1px solid black",
        "margin": 10,
        "padding": 25,
        "width": 200,
      }
    }
  >
    <button
      className="sc-AxiKw sc-AxheI jAEfKR"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Primary Button
      </span>
    </button>
    <button
      className="sc-AxiKw sc-Axmtr gTSmqz"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Default Button
      </span>
    </button>
    <button
      className="sc-AxiKw sc-AxmLO juzGLJ"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Dashed Button
      </span>
    </button>
    <button
      className="sc-AxiKw sc-fzozJi cEXLFa"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Text Button
      </span>
    </button>
    <button
      className="sc-AxiKw sc-fzoLsD ioZmlg"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span>
        Link Button
      </span>
    </button>
  </div>,
  <div>
    <button
      className="sc-AxiKw sc-Axmtr evtuNb"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span
        className="sc-AxhCb iTZsUI"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
    </button>
    <button
      className="sc-AxiKw sc-Axmtr hlgLWj"
      onClick={[Function]}
      onMouseDown={[Function]}
      shape="circle"
    >
      <span
        className="sc-AxhCb iTZsUI"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
    </button>
    <button
      className="sc-AxiKw sc-Axmtr kdIUgf"
      onClick={[Function]}
      onMouseDown={[Function]}
      shape="round"
    >
      <span
        className="sc-AxhCb iTZsUI"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
    </button>
  </div>,
  <div>
    <button
      className="sc-AxiKw sc-Axmtr dwFjtu"
      onClick={[Function]}
      onMouseDown={[Function]}
      size="lg"
    >
      <span
        className="sc-AxhCb iTZsUI"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
      <span>
        Trash
      </span>
    </button>
    <button
      className="sc-AxiKw sc-Axmtr evtuNb"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <span
        className="sc-AxhCb iTZsUI"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
      <span>
        Trash
      </span>
    </button>
    <button
      className="sc-AxiKw sc-Axmtr kBhEii"
      onClick={[Function]}
      onMouseDown={[Function]}
      size="sm"
    >
      <span
        className="sc-AxhCb iTZsUI"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
      <span>
        Trash
      </span>
    </button>
  </div>,
]
`;
